docs/user/guides/build/skip-build.rst

```diff
@@ -0,0 +1,230 @@
+         # This is a special exit code on Read the Docs that will cancel the build immediately.
+         - |
+           if [ "$READTHEDOCS_VERSION_TYPE" = "external" ] && git diff --quiet origin/main -- docs/ .readthedocs.yaml;
+           then
+             exit 183;
+           fi
+
+You can customize this example by:
+
+* Adding more paths to check: ``docs/ .readthedocs.yaml requirements/docs.txt``
+* Checking against a different branch: ``origin/develop`` instead of ``origin/main``
+* Using different comparison operators to check for specific file patterns
+
+Skip builds based on commit message
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example skips builds when the commit message contains ``[skip ci]`` or ``[ci skip]``:
+
+.. code-block:: yaml
+   :caption: .readthedocs.yaml
+
+   version: 2
+   build:
+     os: "ubuntu-22.04"
+     tools:
+       python: "3.12"
+     jobs:
+       post_checkout:
+         # Use `git log` to check if the latest commit contains "skip ci",
+         # in that case exit the command with 183 to cancel the build
+         - (git --no-pager log --pretty="tformat:%s -- %b" -1 | paste -s -d " " | grep -viq "skip ci") || exit 183
+
+This pattern is commonly used in CI/CD systems to skip builds for administrative commits,
+such as version bumps or documentation typos.
+
+Skip builds for specific branch patterns
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example skips builds for branches that match certain patterns,
+such as personal development branches:
+
+.. code-block:: yaml
+   :caption: .readthedocs.yaml
+
+   version: 2
+   build:
+     os: "ubuntu-22.04"
+     tools:
+       python: "3.12"
+     jobs:
+       post_checkout:
+         # Skip builds for branches starting with "dev/" or "experiment/"
+         - |
+           if echo "$READTHEDOCS_GIT_IDENTIFIER" | grep -qE "^(dev|experiment)/"; then
+             exit 183;
+           fi
+
+Skip builds based on file types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example skips builds when all changes are to non-documentation files,
+such as only images or data files:
+
+.. code-block:: yaml
+   :caption: .readthedocs.yaml
+
+   version: 2
+   build:
+     os: "ubuntu-22.04"
+     tools:
+       python: "3.12"
+     jobs:
+       post_checkout:
+         # Skip if only non-documentation files changed (e.g., only images or data files)
+         - |
+           if [ "$READTHEDOCS_VERSION_TYPE" = "external" ]; then
+             # Get list of changed files
+             CHANGED_FILES=$(git diff --name-only origin/main)
+             # Check if any changed files are documentation-related.
+             # If ALL files are non-documentation (grep finds no documentation files),
+             # then we skip the build by exiting with 183.
+             if ! echo "$CHANGED_FILES" | grep -qE "\.(rst|md|py|yaml|yml|txt|toml)$"; then
+               exit 183;
+             fi
+           fi
+
+Best practices
+--------------
```

Comment:
This section also feels unnecessary. The impact of not generating previews and use of env vars seen useful, though